File: src/EzBud.Domain/Category/CategoryGroup.cs
namespace EzBud.Domain.Category;

public class CategoryGroup(Guid id, string name, bool isIncomeGroup = false)
{
    public Guid Id { get; set; } = id;
    public string Name { get; set; } = name;
    public bool IsIncomeGroup { get; set; } = isIncomeGroup;
    public ICollection<Category> Categories { get; set; } = new List<Category>();
}

File: src/main/java/dk/michaelbui/ezbud_server/domain/model/CategoryGroup.java
package dk.michaelbui.ezbud_server.domain.model;

import java.util.ArrayList;
import java.util.Collection;
import java.util.UUID;

public class CategoryGroup {
    private UUID id = UUID.randomUUID();
    private String name;
    private boolean isIncomeGroup;
    private Collection<Category> categories = new ArrayList<>();

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isIncomeGroup() {
        return isIncomeGroup;
    }

    public void setIncomeGroup(boolean incomeGroup) {
        isIncomeGroup = incomeGroup;
    }

    public Collection<Category> getCategories() {
        return categories;
    }

    public void setCategories(Collection<Category> categories) {
        this.categories = categories;
    }
}
